src/pip/_internal/commands/show.py

```diff
@@ -122,47 +148,92 @@ def search_packages_info(query):
         yield package
 
 
-def print_results(distributions, list_files=False, verbose=False):
+def get_package_info(dist, list_files=False, verbose=False):
+    """
+    Gather details from an installed distribution.
+    """
+    name = dist.get('name', '')
+    required_by = [
+        pkg.project_name for pkg in pkg_resources.working_set
+        if name in [required.name for required in pkg.requires()]
+    ]
+    info = [
+        ("Name", name),
+        ("Version", dist.get('version', '')),
+        ("Summary", dist.get('summary', '')),
+        ("Home-page", dist.get('home-page', '')),
+        ("Author", dist.get('author', '')),
+        ("Author-email", dist.get('author-email', '')),
+        ("License", dist.get('license', '')),
+        ("Location", dist.get('location', '')),
+        ("Requires", dist.get('requires', [])),
+        ("Required-by", required_by)
+    ]
+
+    if verbose:
+        classifiers = [str(classifier)
```

Comment:
Do we really need a `str()` ? What else does `dist.get('classifiers', [])` contain ?

Reply:
OK. I've removed similar list comp for 'files'.